src/typedefs.ts

```diff
@@ -100,6 +95,8 @@ export type TCacheCanvasDimensions = {
   y: number;
 };
 
+export type TBounds = IPoint[];
```

Comment:
no IPoint is used

Reply:
I'm keeping this for two reasons: IPoint is just a generalized version of Point, and the bounds might be more than a pair (think general polygon bounds)

Reply:
It is not used anywhere else. 
TBounds is not needed at all. I want it removed. Just adding confusion to naming. 
No one will use bounds of curve in other cases because we have stroke projection for that
Please remove it. The method understand the return type so you only need to type 
`boundsOfCurveCache: Record<string, [Point,Point]> = {};` and remove the rest

Reply:
and perhaps add `as const` to the bounds
```
  const result = [
    new Point(Math.min(...bounds[0]), Math.min(...bounds[1])),
    new Point(Math.max(...bounds[0]), Math.max(...bounds[1])),
  ] as const;
```

Reply:
I went ahead and re did the type for the bounds to `TRectBounds = [min: IPoint, max: IPoint]`. I was going to remove it completely but then I realized that I'd need to define it twice (once for the `boundsOfCurve` in the cache and once for the function return signature) which is just bad practice. I think this is probably the best compromise.

Reply:
I'm going to close this convo. Feel free to reopen if necessary

Reply:
I an live with this now though if you use:


> and perhaps add `as const` to the bounds
> 
> ```
>   const result = [
>     new Point(Math.min(...bounds[0]), Math.min(...bounds[1])),
>     new Point(Math.max(...bounds[0]), Math.max(...bounds[1])),
>   ] as const;
> ```

you don't need to define the return type or `result` at all and you are left with defining boundsOfCurves only.

Reply:
Sorry for the delay, I finally had time to revisit this. When you do `as const` it infers the type as `readonly [Point, Point]` not `[IPoint, IPoint]` like we actually want. I could change the signature to `readonly [Point, Point]` but that's unnecessarily strict. I checked the code and it being a `Point` specifically is not relevant, since the `Point` methods aren't used, so `IPoint` is definitely the right type in this case. I'm not going to touch it but if it irks you I don't actually have a strong opinion so feel free to change it. I just want to be done with this PR, haha.

Reply:
it is good enough